plugins/modules/ec2_vpc_subnet.py

```diff
@@ -279,6 +285,9 @@ def create_subnet(conn, module, vpc_id, cidr, ipv6_cidr=None, az=None, start_tim
     if az:
         params['AvailabilityZone'] = az
 
+    if outpost_arn:
```

Comment:
@mandar242 you can use the same outpost validation function you used for `ec2_vol` module  here.